src/simularium/versionHandlers.ts

```diff
@@ -1,27 +1,94 @@
+import { mapValues } from "lodash";
 import * as si from "si-prefix";
 
 import { DEFAULT_CAMERA_SPEC } from "../constants";
 import {
+    CubeDisplayType,
+    GizmoDisplayType,
+    ObjDisplayType,
+    PdbDisplayType,
+    SphereDisplayType,
     TrajectoryFileInfo,
     TrajectoryFileInfoAny,
     TrajectoryFileInfoV1,
 } from "./types";
+import { AgentDisplayDataWithGeometry } from "./VisGeometry";
 
+// the data may come in missing any of these values
+export interface AgentTypeVisDataPreProcessing {
+    displayType?:
+        | PdbDisplayType
+        | ObjDisplayType
+        | SphereDisplayType
+        | CubeDisplayType
+        | GizmoDisplayType;
+    url?: string;
+    color?: string;
+}
+
+export interface AgentDisplayDataPreProcessing {
+    name: string;
+    geometry?: AgentTypeVisDataPreProcessing;
+}
+export interface EncodedTypeMappingPreProcessing {
+    [key: number]: AgentDisplayDataPreProcessing;
+}
 /*
 Handles different trajectory file format versions.
 Currently supported versions: 1, 2
 */
-const LATEST_VERSION = 2;
+const LATEST_VERSION = 3;
 const VERSION_NUM_ERROR = "Invalid version number in TrajectoryFileInfo:";
 
+export const sanitizeAgentMapGeometryData = (
+    typeMapping: EncodedTypeMappingPreProcessing
+): { [key: number]: AgentDisplayDataWithGeometry } => {
+    return mapValues(typeMapping, (value: AgentDisplayDataPreProcessing) => {
+        let geometry = {};
+        if (value.geometry) {
+            let url = value.geometry.url || "";
+            let { displayType } = value.geometry;
+            if (!displayType) {
+                // we're relying on the data to have a displayType to tell us what sort of data the url is pointing at
+                // if the user fails to provide the displayType, we'll default to loading a sphere, and clear out the url
+                url = "";
```

Comment:
is this a good place to console.log?  since you are destroying any provided url, it might be a good chance to print the url here at this time.

Reply:
you mean to display to the user?

Reply:
yes, to the user or to the programmer who is wondering why the heck the geometry looks wrong

Reply:
I added a response if the url gets cleared out. Also based on the acceptance criteria, I added another case where the geo defaults to spheres, and alerts the user about it.